### PR TITLE
LE Audio: LC3 macro hardcode fix

### LINKED ISSUE #43202: LE Audio: Avoid hardcoding context type for LC3 macros
**Is your enhancement proposal related to a problem? Please describe.**
The `BT_CODEC_LC3_CONFIG_*` macros have a hardcoded context type, which doesn't really make any sense. 

**Describe the solution you'd like**
I see 3 options:
1. The context type should be removed from `BT_CODEC_LC3_CONFIG_*` and thus also from `BT_CODEC_LC3_CONFIG`
2. The context type becomes a parameter to `BT_CODEC_LC3_CONFIG_*`
3. The `BT_CODEC_LC3_CONFIG_*` should be removed and users should use `BT_CODEC_LC3_CONFIG` directly

**Describe alternatives you've considered**
See above

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by Thalley**
Updates a bunch of LC3 macros from using hardcoded context and location to be more flexible. 

fixes https://github.com/zephyrproject-rtos/zephyr/issues/43202

### CODE REVIEW

**Review by alwa-nordic — DISMISSED** [review]

**Comment by Thalley** [comment]
Marking this PR as draft, as I've yet to find a solution to the odd documentation builds error:
```
 Warning, treated as error:
/home/runner/work/zephyr/zephyr/include/zephyr/bluetooth/audio/lc3.h:315: argument 'em' of command @param is not found in the argument list of BT_CODEC_LC3_CONFIG(_freq, _duration, _loc, _len, _frames_per_sdu, _stream_context)
```

**Comment by Thalley** [comment]
The documentation seems to fail to build HTML due to the `_` prefix in the macro parameters (removing them fixes it). Not quite how to fix that in this PR, without removing the `_` prefix...

**Comment by Thalley** [comment]
Rebased without changes

File: subsys/bluetooth/shell/audio.c [hunk 1]
```diff
@@ -54,75 +57,75 @@ struct named_lc3_preset {
...
-	{"32_2_1",  BT_AUDIO_LC3_UNICAST_PRESET_32_2_1},
-	{"441_1_1", BT_AUDIO_LC3_UNICAST_PRESET_441_1_1},
-	{"441_2_1", BT_AUDIO_LC3_UNICAST_PRESET_441_2_1},
-	{"48_1_1",  BT_AUDIO_LC3_UNICAST_PRESET_48_1_1},
-	{"48_2_1",  BT_AUDIO_LC3_UNICAST_PRESET_48_2_1},
-	{"48_3_1",  BT_AUDIO_LC3_UNICAST_PRESET_48_3_1},
-	{"48_4_1",  BT_AUDIO_LC3_UNICAST_PRESET_48_4_1},
-	{"48_5_1",  BT_AUDIO_LC3_UNICAST_PRESET_48_5_1},
-	{"48_6_1",  BT_AUDIO_LC3_UNICAST_PRESET_48_6_1},
+	{"8_1_1",   BT_AUDIO_LC3_UNICAST_PRESET_8_1_1(LOCATION, CONTEXT)},
```
Comment (Casper-Bonde-Bose) [inline subsys/bluetooth/shell/audio.c:60]: I'm not sure it makes sense to have a compile-time defined location. When setting up a stereo stream you typically want to make two different locations - RIGHT to one stream and LEFT to the other - or pack left and right into the same stream.

**Review by Casper-Bonde-Bose — DISMISSED** [review]
LGTM
The shall application should at some point be made dynamic to support run-time selection of LEFT/RIGHT audio location - but the macros support this.

File: subsys/bluetooth/shell/audio.c (hunk 1 above)
Comment (Thalley) [reply subsys/bluetooth/shell/audio.c:60]: Agreed :)

**Review by Casper-Bonde-Bose — APPROVED** [review]
LGTM

**Review by asbjornsabo — APPROVED** [review]
LGTM

**Pull request merged**